vnext/Shared/ChakraCoreRuntime.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ChakraRuntime.h"
+
+#include "ChakraCore.h"
+#include "ChakraCoreDebugger.h"
+
+namespace Microsoft::JSI {
+
+class ChakraCoreRuntime : public ChakraRuntime {
+ public:
+  ChakraCoreRuntime(ChakraRuntimeArgs &&args) noexcept;
+
+  ~ChakraCoreRuntime() noexcept;
+
+#pragma region ChakraRuntime
+
+  void setupNativePromiseContinuation() noexcept override;
+
+  void startDebuggingIfNeeded() override;
+
+  void stopDebuggingIfNeeded() override;
+
+  std::unique_ptr<const facebook::jsi::Buffer> generatePreparedScript(
+      const std::string &sourceURL,
+      const facebook::jsi::Buffer &sourceBuffer) noexcept override;
+
+  facebook::jsi::Value evaluateJavaScriptSimple(const facebook::jsi::Buffer &buffer, const std::string &sourceURL)
+      override;
+
+  bool evaluateSerializedScript(
+      const facebook::jsi::Buffer &scriptBuffer,
+      const facebook::jsi::Buffer &serializedScriptBuffer,
+      const std::string &sourceURL) override;
+
+#pragma endregion
+
+ private:
+  // core-only??
```

Comment:
>// core-only?? [](start = 2, length = 14)

ChakraCore and system Chakra use different debuggers. So, the debugger initializaiton could be a little bit different between the two. It wonder if it is soething that can be abstracted between them at some level.

Reply:
From the original implementation, it looks like the debugging classes for System Chakra were defined as stubs (i.e. `class DebugProtocolHandler {};`.